crates/uv-globfilter/src/glob_dir_filter.rs

```diff
@@ -0,0 +1,277 @@
+use bstr::{ByteSlice, ByteVec};
+use globset::{Glob, GlobSet, GlobSetBuilder};
+use regex_automata::dfa;
+use regex_automata::dfa::Automaton;
+use std::path::{Path, MAIN_SEPARATOR, MAIN_SEPARATOR_STR};
+use tracing::warn;
+
+/// Chosen at a whim -Konsti
+const DFA_SIZE_LIMIT: usize = 1_000_000;
```

Comment:
This is probably fine. Or at least, as a whim, I don't see this being a huge issue, especially given the restricted glob syntax.